keyboards/ploopyco/trackball_nano/keymaps/maddie/keymap.c

```diff
@@ -18,39 +18,24 @@
  */
 #include QMK_KEYBOARD_H
 
+// Configuration options
+#define SCROLL_TIMEOUT 25
+
 // safe range starts at `PLOOPY_SAFE_RANGE` instead.
-uint8_t scroll_enabled = 0;
-uint8_t lock_state = 0;
-int8_t  delta_x        = 0;
-int8_t  delta_y        = 0;
+bool scroll_enabled = 0;
+bool lock_state     = 0;
 
 // Dummy
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {{{ KC_NO }}};
 
-void process_mouse_user(report_mouse_t *mouse_report, int8_t x, int8_t y) {
+report_mouse_t pointing_device_task_user(report_mouse_t mouse_report) {
     if (scroll_enabled) {
-        delta_x += x;
-		delta_y += y;
-
-		if (delta_x > 60) {
-			mouse_report->h = 1;
-			delta_x = 0;
-		} else if (delta_x < -60) {
-			mouse_report->h = -1;
-			delta_x = 0;
-		}
-
-		if (delta_y > 15) {
-			mouse_report->v = -1;
-			delta_y = 0;
-		} else if (delta_y < -15) {
-			mouse_report->v = 1;
-			delta_y = 0;
-		}
-    } else {
-        mouse_report->x = x;
-		mouse_report->y = y;
```

Comment:
the `delta_` code should remain - this is to reduce the number of scroll events to a usable level.

Reply:
When I had the `delta_` code in there, I would get small but visible mouse movement along with scrolling, so I think it needs to be modified, but I'm not sure how exactly.

Reply:
@hexadecatrienoic I managed to figure out how to restore the `delta_` logic while eliminating the mouse movement I was seeing.